packages/plugins/rpc-manager/src/lib/index.ts

```diff
@@ -20,25 +22,78 @@ export default class RpcManager {
   private events: EmbarkEvents;
   public _accounts: any[] | null = null;
   public _nodeAccounts: any[] | null = null;
+  private blockchainConfig: EmbarkConfig["blockchainConfig"];
   constructor(private readonly embark: Embark) {
     this.events = embark.events;
     this.logger = embark.logger;
     this.rpcModifierEvents = new Events() as EmbarkEvents;
+    this.blockchainConfig = this.embark.config.blockchainConfig;
     this.init();
   }
 
-  private async init() {
+  protected get web3() {
+    return (async () => {
+      if (!this._web3) {
+        let provider;
+        let testsCliOptions;
+        let isTest = false;
 
-    this.rpcModifierEvents.setCommandHandler("nodeAccounts:updated", this.updateAccounts.bind(this));
-    this.rpcModifierEvents.setCommandHandler("nodeAccounts:added", async (addedNodeAccount: any, cb: Callback<null>) => {
+        await this.events.request2("blockchain:started");
+
+        if (this.embark.currentContext.includes(constants.contexts.test)) {
+          isTest = true;
+          testsCliOptions = await this.events.request2("tests:cli:options");
+        }
+        if (isTest && testsCliOptions.node === constants.blockchain.vm) {
```

Comment:
For now, you should check the the `blockchainConfig.clent === 'vm'` (at least until my Ganache plugin PR is merged), since the CLI options are not the real truth. See comment below for more context (sorry, I wrote my comments in a weird order)